### PR TITLE
Feat: Add `ElasticConfig` message type for torch elastic training

### PR DESCRIPTION
**Pull request by fg91**
# TL;DR
This PR adds a custom proto for a torch elastic config for distributed training using [torchrun](https://pytorch.org/docs/stable/elastic/run.html). The elastic configuration set by the user in the task decorator is converted to the [`ElasticPolicy` in kubeflow's `PytorchJob`](https://github.com/kubeflow/training-operator/blob/8a066f9386e9fb739815e9001f4fe53158227646/pkg/apis/kubeflow.org/v1/pytorch_types.go#L90).

See https://github.com/flyteorg/flyte/issues/3614 issue for more details.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3614

### CODE REVIEW

File: protos/flyteidl/plugins/pytorch.proto [hunk 1]
```diff
@@ -4,8 +4,22 @@ package flyteidl.plugins;
 
 option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins";
 
+// Custom proto for torch elastic config for distributed training using 
+// https://github.com/kubeflow/training-operator/blob/master/pkg/apis/kubeflow.org/v1/pytorch_types.go
+message ElasticConfig {
+  string rdzv_backend = 1;
+  int32 min_replicas = 2;
```
Comment (kumare3) [inline protos/flyteidl/plugins/pytorch.proto:11]: Do we just ignore workers if this is set?

Comment (fg91) [reply protos/flyteidl/plugins/pytorch.proto:11]: No, we still set it, see [here](https://github.com/flyteorg/flyteplugins/blob/b644427ef3af6ccb9b6771cdfa6191b2a6209e35/go/tasks/plugins/k8s/kfoperators/pytorch/pytorch.go#L97). In the kubeflow `PyTorchJob` the number of `replicas` is set outside of the `ElasticPolicy` (see [here](https://github.com/kubeflow/training-operator/blob/8a066f9386e9fb739815e9001f4fe53158227646/examples/pytorch/elastic/echo/echo.yaml#L13)) where we also set it for the non-elastic pytorch job.

**Review by kumare3 — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
